msal/src/test/java/com/microsoft/identity/client/e2e/tests/mocked/SingleAccountAcquireTokenMockTest.java

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.client.e2e.tests.mocked;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.microsoft.identity.client.IAccount;
+import com.microsoft.identity.client.IPublicClientApplication;
+import com.microsoft.identity.client.ISingleAccountPublicClientApplication;
+import com.microsoft.identity.client.e2e.utils.RoboTestUtils;
+import com.microsoft.identity.client.exception.MsalException;
+
+import static com.microsoft.identity.client.e2e.utils.TestConstants.Configurations.SINGLE_ACCOUNT_MODE_AAD_CONFIG_FILE_PATH;
+import static org.junit.Assert.fail;
+
+public class SingleAccountAcquireTokenMockTest extends AcquireTokenMockTest {
+
+    @Override
+    public String getConfigFilePath() {
+        return SINGLE_ACCOUNT_MODE_AAD_CONFIG_FILE_PATH;
+    }
+
+    @Override
+    IAccount performGetAccount(IPublicClientApplication application, String loginHint) {
+        final IAccount[] requestedAccount = {null};
+        final ISingleAccountPublicClientApplication singleAcctApp = (ISingleAccountPublicClientApplication) application;
+        singleAcctApp.getCurrentAccountAsync(new ISingleAccountPublicClientApplication.CurrentAccountCallback() {
+            @Override
+            public void onAccountLoaded(@Nullable IAccount activeAccount) {
+                if (activeAccount != null) {
+                    requestedAccount[0] = activeAccount;
+                } else {
+                    fail("No account found");
+                }
+            }
+
+            @Override
+            public void onAccountChanged(@Nullable IAccount priorAccount, @Nullable IAccount currentAccount) {
+                if (currentAccount != null) {
```

Comment:
For non-FLW scenario, this is not expected to be invoked.
It should be invoked if the account was 'changed' by other apps.

I think we can probably just put 'fail' here.

Reply:
It is being invoked actually

Reply:
I'll address this in a future PR